open_spiel/python/algorithms/efr_test.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2023 DeepMind Technologies Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for open_spiel.python.algorithms.efr."""
+
+from absl.testing import absltest
+from absl.testing import parameterized
+import numpy as np
+
+from open_spiel.python import policy
+from open_spiel.python.algorithms import efr
+from open_spiel.python.algorithms import expected_game_score
+import pyspiel
+
+_KUHN_GAME = pyspiel.load_game("kuhn_poker")
+_LEDUC_GAME = pyspiel.load_game("leduc_poker")
+
+_KUHN_UNIFORM_POLICY = policy.TabularPolicy(_KUHN_GAME)
+_LEDUC_UNIFORM_POLICY = policy.TabularPolicy(_LEDUC_GAME)
+_DEVIATIONS_ = ["blind action", "informed action", "blind cf", "informed cf", "bps", "cfps", "csps", "tips", "bhv"]
+
+class EFRTest(parameterized.TestCase, absltest.TestCase):
```

Comment:
Since this is EFR, can we also have a non-zero sum game and multi-player game?

We've exposed some of the distance to the various *CE and *CCE algorithm. I just noticed that we don't have examples of using them, though, I'll add some and let you know.

Reply:
I added some examples of checking distance to CCE here: https://github.com/google-deepmind/open_spiel/commit/8859b79f895ce7bddaf62d77ba6c0946956312ea. Should be easy to expose the other ones (EF(C)CE, AF(C)CE) if you want to try them too. 

Would be good to have at least one graph where we confirm EFR approaching is CE or CCE when it should be, and then a test on a smallish game where this distance is decreasing (over a small number of iterations so the test doesn't take too long -- see the CFR tests I added).

Reply:
Amazing, thanks for the suggestions, I'll work on this over the next few days and let you know if I run into any issues.

Reply:
I have ran a test on 3 player Kuhn Poker for all deviation types over 1000 iterations and plotted the distance to the CCE.
![CCE restricted](https://github.com/google-deepmind/open_spiel/assets/13368287/e9fc3abd-baaa-412e-b601-84299b12a4c1)
![CCE full deviations](https://github.com/google-deepmind/open_spiel/assets/13368287/5ac9ce08-a812-4257-b55e-8d954a7a431d)
 I think it roughly checks out, let me know what you think!